stackstate_checks_base/stackstate_checks/base/utils/http_helper.py

```
@@ -0,0 +1,786 @@
+        Functionality:
+            - Create and maintain the connection values outside of the Session() and Request() objects
+            - SET && GET HTTP Timeout
+            - SET && GET HTTP Retry Policy
+            - SET && GET HTTP SSL Verification
+            - Sending the HTTP Request and Session to the Endpoint
+
+    """
+
+    _response_types = [item.value for item in HTTPResponseType]
+
+    log = logging.getLogger('%s.response_handler' % __name__)
+
+    def __init__(self, response_model=None):
+        self._response_model = HTTPHelperResponseModel(response_model) if response_model is not None \
+            else HTTPHelperResponseModel()
+
+    def get_status_code(self):
+        """
+        Return the status code response from the executed Session() object
+        """
+        return self._response_model.response.status_code
+
+    def get_json(self):
+        """
+        First we attempt to decode the json then we run validation on the data.
+        This will then return valid decoded json from the execute Session() object
+        """
+        # We first decode the response with the encoding header
+        decoded_response = self._response_model.response.content.decode(
+            self._response_model.response.encoding
+        )
+
+        ensured_string_values = to_string(decoded_response)
+
+        # Next we use the json load function with its build-in unicode functionality to decode the object
+        # This will work on straight text or json objects
+        # This should take the object from a unicode state to a dict state
+        json_decoded_response = json.loads(ensured_string_values)
+
+        # If valid json was found return this json
+        # Run validation before we return the value
+        if issubclass(type(json_decoded_response), dict) and self.validate_body_schematic(json_decoded_response):
+            return json_decoded_response
+        else:
+            return None
```

This is a silent failure. Under what circumstances is this a correct answer? Raise an exception.

---

I usually prefer the Some None approach instead of forcing errors if possible